Build-qt.sh

```diff
@@ -0,0 +1,180 @@
+#!/bin/bash
+
+show_help() {
+cat << EOF
+Usage: ${0##*/} [-h] [-c] [-j] [-q QT_INSTALL_DIR] [-m CMAKE]
+
+This script is a convenience script to install Qt on the system. It:
+
+- Auto-detect if running on MacOS or Linux
+- Compiles zlib.
+- Compiles openssl.
+- Compiles and install qt.
+
+Options:
+
+  -h             Display this help and exit.
+  -c             Clean directories that are going to be used.
+  -q             Installation directory for Qt.
+  -m             Path for cmake.
+  -j             Number of threads to compile tools
+EOF
+}
+clean=0
+nbthreads=1
+while [ $# -gt 0 ]; do
+  case "$1" in
+    -h)
+      show_help
+      exit 0
+      ;;
+    -c)
+      clean=1
+      ;;
+    -q)
+      install_dir=$2
+      shift
+      ;;
+    -m)
+      cmake=$2
+      shift
+      ;;
+    -j)
+      nbthreads=$2
+      shift
+      ;;
+    *)
+      show_help >&2
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+# If "clean", remove all directories and temporary files
+# that are downloaded and used in this script.
+if [ $clean -eq 1 ]
+then
+  echo "Remove previous files and directories"
+  rm -rf zlib*
+  rm -f openssl-1.0.1h.tar.gz
+  rm -rf openssl-1.0.1h
+  rm -f qt-everywhere-opensource-src-4.8.7.tar.gz
+  rm -rf qt-everywhere-opensource-src-4.8.7
+  rm -rf qt-everywhere-opensource-build-4.8.7
+fi
+# If cmake path was not given, verify that it is available on the system
+# CMake is required to configure zlib
+if [[ -z "$cmake" ]]
+then
+  cmake=`which cmake`
+  if [ $? -ne 0 ]
+  then
+    echo "cmake not found"
+    exit 1
+  fi
+  echo "Using cmake found here: $cmake"
+fi
+
+# Download archives (Qt, and openssl
+echo "Download openssl"
+curl -OL https://packages.kitware.com/download/item/6173/openssl-1.0.1h.tar.gz
+echo "Download Qt"
+curl -OL http://download.qt.io/official_releases/qt/4.8/4.8.7/qt-everywhere-opensource-src-4.8.7.tar.gz
+
+# Check if building on MacOS or Linux
+# And verifies downloaded archives accordingly
+if [ "$(uname)" == "Darwin" ]
+then
+  # MacOS
+  deployment_target=10.6
+  sysroot=/Developer/SDKs/MacOSX10.6.sdk
```

Comment:
deployment_target, sysroot, and architecture are now options.  It would be indeed great to adapt the CMake code you are mentioning above, but I don't have a Mac to test the script...